packages/core/src/utils/index.ts

```diff
@@ -264,3 +265,19 @@ export function stringify(val: unknown) {
     return value;
   });
 }
+
+/**
+ * Calculate the greatest common divisor (GCD) of two Num values using the Euclidean algorithm.
+ *
+ * @param a - First operand.
+ * @param b - Second operand.
+ * @returns GCD(a, b) as a Num.
+ */
+export function gcd(a: Num, b: Num): Num {
```

Comment:
I think we should make the param type `NumLike` if we want to make `gcd` public. Otherwise, we should make it private or advanced.

Reply:
Done